lib/cashaddr.py

```diff
@@ -118,7 +119,7 @@ def _decode_payload(addr):
         raise ValueError('address prefix is missing: {}'.format(addr))
     if not all(33 <= ord(x) <= 126 for x in prefix):
         raise ValueError('invalid address prefix: {}'.format(prefix))
-    if not (8 <= len(payload) <= 124):
+    if not (8 <= len(payload) <= 256):
```

Comment:
??  CashAddr payloads have a well specified length.  Why was this file modified at all?

Reply:
If you must do this pass a flag own to use a different value for your special case.  I would prefer if you didn't have to touch this file.. this is bizarre to me that you would have to.... Why?

Reply:
The cashaddr code doesn't seem to work out of the box for "addresses" as long as the paycodes.  There were two problems encountered, both of which appear to be sanity check related.  The one here (124) was bumped from the 80 value of the original bech32.   here->https://github.com/Electron-Cash/Electron-Cash/commit/ae60cc9386f1082421a3bddb41a1baa94ce4edcd#diff-6624c8abadfc8d6860e0879af63e447b

The other sanity check is related to length but causes paycode to fail and doesn't seem necessary.  But this is why I created a flag (rpa_paycode = False).  

I can make them both dependent on a flag or just separate out the functions.  Either way(s) seem ok to me.

Reply:
Again, why change the default execution path for your custom use-case?

There will be unintended side effects.  This change now impacts the app globally --

You already have that flag you are passing .. please pass it down and default the normal execution path without the flag set to the old behavior in all cases in this file, minimally.

Reply:
Also -- how did you arrive at a 256 value?  Is that a magical random number you chose because it's a power of two and "looks right"?

Can you please motivate this further? (That is: talk about it and explain).

Reply:
My thought was that if this is a kind of arbitrary value, then maybe it's better to just increase it rather than start adding custom rules...and the 256 was arbitrary as well.  If you prefer just to add a custom flag and pass it down, can do!

Reply:
How do you know that it's arbitrary, though?  It seems deliberately chosen.  Do you have anything I can look at to see that?

Reply:
I don't know for sure.  I had asked Shammah about that commit and his reply seemed to indicate that.

In any case, I will implement a flag so the default behavior is not impacted.  I will also investigate "encoding a version properly into the payload to make the length check succeed"  as you recommended.